src/pages/home/report/comment/TextCommentFragment.js

```diff
@@ -0,0 +1,118 @@
+import Str from 'expensify-common/lib/str';
+import PropTypes from 'prop-types';
+import React, {memo} from 'react';
+import Text from '@components/Text';
+import withLocalize, {withLocalizePropTypes} from '@components/withLocalize';
+import withWindowDimensions, {windowDimensionsPropTypes} from '@components/withWindowDimensions';
+import ZeroWidthView from '@components/ZeroWidthView';
+import compose from '@libs/compose';
+import convertToLTR from '@libs/convertToLTR';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import * as EmojiUtils from '@libs/EmojiUtils';
+import reportActionFragmentPropTypes from '@pages/home/report/reportActionFragmentPropTypes';
+import reportActionSourcePropType from '@pages/home/report/reportActionSourcePropType';
+import editedLabelStyles from '@styles/editedLabelStyles';
+import styles from '@styles/styles';
+import themeColors from '@styles/themes/default';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import RenderCommentHTML from './RenderCommentHTML';
+
+const propTypes = {
+    /** The reportAction's source */
+    source: reportActionSourcePropType.isRequired,
+
+    /** The message fragment needing to be displayed */
+    fragment: reportActionFragmentPropTypes.isRequired,
+
+    /** Should this message fragment be styled as deleted? */
+    styleAsDeleted: PropTypes.bool.isRequired,
+
+    /** Text of an IOU report action */
+    iouMessage: PropTypes.string,
+
+    /** Should the comment have the appearance of being grouped with the previous comment? */
+    displayAsGroup: PropTypes.bool.isRequired,
+
+    /** Additional styles to add after local styles. */
+    style: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]).isRequired,
+
+    ...windowDimensionsPropTypes,
+
+    /** localization props */
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    iouMessage: undefined,
+};
+
+function TextCommentFragment(props) {
+    const {fragment, styleAsDeleted} = props;
+    const {html, text} = fragment;
```

Comment:
Might be related: When there is a mention and emoji the backend only sending emoji itself on `text` and it is getting treated as only emoji and enlarged in chat causing UI problems. https://github.com/Expensify/App/issues/50306